frontend/src/services/web3-provider/metamask-provider.ts

```diff
@@ -67,7 +67,12 @@ export class MetaMaskProvider extends InjectedProvider implements ISigner {
       await this.web3Provider.send('wallet_switchEthereumChain', [{ chainId: newChainIdHex }]);
       return true;
     } catch (error) {
-      if ((error as { code: number })?.code === unrecognizedChainErrorCode) {
+      type MetaMaskError = { code: number };
+      type MetaMaskMobileError = { data?: { originalError?: { code: number } } };
+      if (
+        (error as MetaMaskError)?.code === unrecognizedChainErrorCode ||
+        (error as MetaMaskMobileError)?.data?.originalError?.code === unrecognizedChainErrorCode
+      ) {
```

Comment:
Can we separate this checks into two functions? Currently its really hard to read and understand whats going on here.

The function can be like:
```
type MetaMaskError = { code: number };
function isMetaMaskError(error: any, code: number): error is MetaMaskError {
  const mmError = error as MetaMaskError;
  return !!mmError.code && mmError.code === code;
}
```

Reply:
I think an `isMetaMaskError` function should not check for a specific code to be present. I added a `hasErrorCode`. Please have look at the fixup commit.

Reply:
Yeah, that looks even better! Thanks